src/Nordea/Components/Button.elm

```diff
@@ -63,58 +71,65 @@ type Variant
     | Card
 
 
-type alias Config =
-    { variant : Variant, styles : List Style }
+type alias Config msg =
+    { variant : Variant
+    , styles : List Style
+    , htmlTag : List (Attribute msg) -> List (Html msg) -> Html msg
+    }
 
 
-type Button
-    = Button Config
+type Button msg
+    = Button (Config msg)
 
 
-init : Variant -> Button
+init : Variant -> Button msg
 init variant =
     Button
-        { variant = variant, styles = [] }
+        { variant = variant, styles = [], htmlTag = Html.button }
 
 
-primary : Button
+primary : Button msg
 primary =
     init Primary
 
 
-secondary : Button
+secondary : Button msg
 secondary =
     init Secondary
 
 
-tertiary : Button
+tertiary : Button msg
 tertiary =
     init Tertiary
 
 
-card : Button
+card : Button msg
 card =
     init Card
 
 
-withStyles : List Style -> Button -> Button
-withStyles styles (Button config) =
-    Button { config | styles = styles }
-
-
 
 -- VIEW
 
 
-view : List (Attribute msg) -> List (Html msg) -> Button -> Html msg
+view : List (Attribute msg) -> List (Html msg) -> Button msg -> Html msg
 view attributes children (Button config) =
-    Html.styled Html.button
+    let
+        variantSpecificChildren =
+            case config.variant of
+                Card ->
+                    [ Icons.arrowRight [ attribute "type" "arrowicon", css [ width (rem 1), height (rem 1) ] ] ]
```

Comment:
Is the `arrowicon` type attribute only used to to apply the movement on hover? If that is the case, I think it is more correct to use a classname for that. Based on [this](https://www.geeksforgeeks.org/html-type-attribute/), it doesnt look like the type is valid on div elements.

```suggestion
                    [ Icons.arrowRight [ class "arrowicon", css [ width (rem 1), height (rem 1) ] ] ]
```